### PR TITLE
Engineer Bot stuff and simple-ipc updates

### PR DESCRIPTION
**Pull request by BenCat07**
~DO NOT MERGE YET; STILL NEEDS SOME TESTING TO BE DONE ON CATBOT-SETUP~

Testing done, everything is fine

### CODE REVIEW

File: src/helpers.cpp [hunk 1]
```diff
@@ -685,6 +685,13 @@ bool GetHitbox(CachedEntity *entity, int hb, Vector &out)
     return true;
 }
 
+void MatrixGetColumn(const matrix3x4_t &in, int column, Vector &out)
```
Comment (TotallyNotElite) [inline src/helpers.cpp]: This is a valve mathlib function.

File: src/hooks/CreateMove.cpp [hunk 2]
```diff
@@ -127,13 +127,7 @@ namespace hooked_methods
...
 DEFINE_HOOKED_METHOD(CreateMove, bool, void *this_, float input_sample_time, CUserCmd *cmd)
 {
-#define TICK_INTERVAL (g_GlobalVars->interval_per_tick)
-#define TIME_TO_TICKS(dt) ((int) (0.5f + (float) (dt) / TICK_INTERVAL))
-#define TICKS_TO_TIME(t) (TICK_INTERVAL * (t))
-#define ROUND_TO_TICKS(t) (TICK_INTERVAL * TIME_TO_TICKS(t))
-    volatile uintptr_t **fp;
-    __asm__ volatile("mov %%ebp, %0" : "=r"(fp));
-    bSendPackets = reinterpret_cast<bool *>(**fp - 8);
+    bSendPackets = reinterpret_cast<bool *>((uintptr_t) __builtin_frame_address(2) - 8);
```
Comment (TotallyNotElite) [inline src/hooks/CreateMove.cpp]: Does this work on clang?

File: src/hooks/CreateMove.cpp [hunk 3]
```diff
@@ -419,6 +400,20 @@ DEFINE_HOOKED_METHOD(CreateMove, bool, void *this_, float input_sample_time, CUs
...
+    if (nolerp && !hacks::shared::backtrack::isBacktrackEnabled)
+    {
+        static const ConVar *pUpdateRate = g_pCVar->FindVar("cl_updaterate");
+        if (!pUpdateRate)
+            pUpdateRate = g_pCVar->FindVar("cl_updaterate");
+        else
+        {
+            // We should adjust cl_interp to be as low as possible
+            if (cl_interp->GetFloat() > 0.152f)
+                cl_interp->SetValue(0.152f);
```
Comment (TotallyNotElite) [inline src/hooks/CreateMove.cpp]: Avoid doing this at all or undo it after nolerp is unset.

**Comment by TotallyNotElite** [comment]
Also m8, have you considered making smaller pull requests on multiple branches instead of one big one?

File: src/hacks/Misc.cpp [hunk 4]
```diff
@@ -224,6 +319,12 @@ void CreateMove()
 // Timer ussr{};
 void DrawText()
 {
+    if (misc_drawhitboxes)
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/Misc.cpp]: True, but that means another Draw Register, but sure i can do that

File: src/hacks/MiscAimbot.cpp [hunk 5]
```diff
@@ -269,12 +273,242 @@ static void SapperAimbot()
         }
     }
 }
+// auto repair
+static settings::Int autorepair_priority{ "autorepair.priority", "0" };
+static settings::Boolean autorepair_enabled("autorepair.enabled", "false");
+static settings::Boolean autorepair_silent("autorepair.silent", "true");
+static settings::Boolean autorepair_autoequip("autorepair.autoequip", "false");
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/MiscAimbot.cpp]: All of them are used

File: .gitmodules [hunk 6]
```diff
@@ -27,4 +27,6 @@
 	url = https://github.com/sakra/cotire
 [submodule "external/clip"]
 	path = external/clip
-	url = https://github.com/nullworks/clip
\ No newline at end of file
+	url = https://github.com/nullworks/clip
+[submodule "external/simple-ipc"]
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply .gitmodules]: Willfix

File: src/PlayerTools.cpp [hunk 7]
```diff
@@ -110,4 +111,32 @@ void onKilledBy(CachedEntity *entity)
...
+        {
+            onKilledBy(ENTITY(killer_id));
+            return;
+        }
+    }
+};
+
+PlayerToolsEventListener &listener()
+{
+    static PlayerToolsEventListener object{};
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/PlayerTools.cpp]: This is simply how we do it for every other occurance of Event listeners

File: src/hacks/Misc.cpp [hunk 8]
```diff
@@ -118,15 +121,107 @@ int getCarriedBuilding()
...
             continue;
-        if (!CE_BYTE(ent, netvar.m_bPlacing))
+        if (!CE_INT(ent, netvar.m_bPlacing))
             continue;
         return i;
     }
     return -1;
 }
+#if ENABLE_VISUALS
+inline void matrix_angles(const matrix3x4_t &matrix, float *angles)
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/Misc.cpp]: Yes but it has different arguments, one of which i can't do anything with: euler_angles

File: src/hacks/MiscAimbot.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/MiscAimbot.cpp]: Wait huh this one really isn't used... i thought i did

File: src/hacks/MiscAimbot.cpp [hunk 9]
```diff
@@ -269,12 +273,242 @@ static void SapperAimbot()
...
+            if (ent->m_Type() != ENTITY_BUILDING)
+                continue;
+
+            // yes, repair enemy buildings
+            if (ent->m_bEnemy())
+                continue;
+
+            float new_distance = g_pLocalPlayer->v_Eye.DistTo(GetBuildingPosition(ent));
+
+            // Further away than old one
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/MiscAimbot.cpp]: I should really rename these variables

File: src/hacks/MiscAimbot.cpp [hunk 10]
```diff
@@ -269,12 +273,242 @@ static void SapperAimbot()
...
+        return;
+
+    if (GetWeaponMode() != weaponmode::weapon_melee)
+        return;
+
+    bool first_run = false;
+    if (autorepair_priority)
+        first_run = true;
+
+    // Run this loop twice if we have priority set
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/MiscAimbot.cpp]: We need to run omce for priority buildings and once for the rest if no priority building was found

File: src/hacks/MiscAimbot.cpp [hunk 11]
```diff
@@ -269,12 +273,242 @@ static void SapperAimbot()
...
+    for (int idx = 0; idx < (*autorepair_priority ? 2 : 1); idx++)
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/MiscAimbot.cpp]: This is just used to determine if we should run the loop once or twice, i should make this more clear

File: src/hacks/Spam.cpp [hunk 12]
```diff
@@ -391,10 +396,34 @@ const std::vector<std::string> builtin_nonecore = { "NULL CORE - REDUCE YOUR RIS
 const std::vector<std::string> builtin_lmaobox  = { "GET GOOD, GET LMAOBOX!", "LMAOBOX - WAY TO THE TOP", "WWW.LMAOBOX.NET - BEST FREE TF2 HACK!" };
 const std::vector<std::string> builtin_lithium  = { "CHECK OUT www.YouTube.com/c/DurRud FOR MORE INFORMATION!", "PWNING AIMBOTS WITH OP ANTI-AIMS SINCE 2015 - LITHIUMCHEAT", "STOP GETTING MAD AND STABILIZE YOUR MOOD WITH LITHIUMCHEAT!", "SAVE YOUR MONEY AND GET LITHIUMCHEAT! IT IS FREE!", "GOT ROLLED BY LITHIUM? HEY, THAT MEANS IT'S TIME TO GET LITHIUMCHEAT!!" };
 
+void teamspam_reload(std::string after)
+{
+    // Clear sp am vector
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/Spam.cpp]: oops

File: src/hacks/Spam.cpp [hunk 13]
```diff
@@ -7,8 +7,10 @@
 
 #include <hacks/Spam.hpp>
 #include <settings/Bool.hpp>
+#include <settings/String.hpp>
 #include "common.hpp"
 #include "MiscTemporary.hpp"
+#include <boost/algorithm/string.hpp>
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/Spam.cpp]: willfix

File: src/helpers.cpp (hunk 1 above)
Comment (BenCat07) [reply src/helpers.cpp]: Yes but we only have headers, i took this from source sdk to like not crash when we try to call it

File: src/hooks/CreateMove.cpp (hunk 3 above)
Comment (BenCat07) [reply src/hooks/CreateMove.cpp]: Can't avoid doing this, and also what if someone just closes their game? Anyways, Okay

**Comment by BenCat07** [comment]
This PR was pending for quite a while with only a few changes

File: src/hacks/Misc.cpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (TotallyNotElite) [reply src/hacks/Misc.cpp]: Why not? Looks pretty straight forward to me. 
https://github.com/ValveSoftware/source-sdk-2013/blob/master/sp/src/public/mathlib/vector.h#L1662 3x vec_t.

File: src/hacks/Misc.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (TotallyNotElite) [reply src/hacks/Misc.cpp]: Just don't name it drawtext. Name it draw or make a function that calls both drawtext and drawhitboxes.

File: src/hooks/CreateMove.cpp (hunk 3 above)
Comment (TotallyNotElite) [reply src/hooks/CreateMove.cpp]: > Can't avoid doing this
I thought nolerp already disables lerp completely, why is this necessary?

File: src/hacks/MiscAimbot.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (TotallyNotElite) [reply src/hacks/MiscAimbot.cpp]: Huh, almost like I read the code I review xd

File: src/hacks/MiscAimbot.cpp (hunk 10 above)
[the remark this answers is not in the document]
Comment (TotallyNotElite) [reply src/hacks/MiscAimbot.cpp]: Could you make that clear in your comment?

File: src/hacks/Misc.cpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (TotallyNotElite) [reply src/hacks/Misc.cpp]: Why not? Looks pretty straight forward to me.
https://github.com/ValveSoftware/source-sdk-2013/blob/master/sp/src/public/mathlib/vector.h#L1662 3x vec_t.

[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/Misc.cpp]: Not sure if they work the same way as a normal valve vector

File: src/hooks/CreateMove.cpp (hunk 3 above)
Comment (BenCat07) [reply src/hooks/CreateMove.cpp]: Well yes but actually no, it disables lerp but only for client, server still thinks you have lerp

File: src/hacks/Misc.cpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (TotallyNotElite) [reply src/hacks/Misc.cpp]: It's literally 3 floats and literally allows construction using QAngles https://github.com/ValveSoftware/source-sdk-2013/blob/master/sp/src/public/mathlib/vector.h#L1668

**Comment by SylveonBottle** [comment]
https://github.com/nullworks/cathook/blob/f76aced3ae1a0c49797be0c98fd69875ca848987/src/hacks/NavBot.cpp#L102

cate

File: src/hacks/Misc.cpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/Misc.cpp]: Alright fair enough, but i'll still have to add that function by hand, so pretty much all i'll be doing is renaming it and possibly moving it elsewhere

File: src/hacks/NavBot.cpp [hunk 14]
```diff
@@ -165,6 +368,315 @@ static bool navToSniperSpot()
...
+    else if (metal >= 100 && !dispenser_built)
+        return Dispenser;
+    return None;
+}
+
+static success_build buildBuilding()
+{
+    int metal = CE_INT(LOCAL_E, netvar.m_iAmmo + 12);
+    // Out of Metal
+    if (metal < 100)
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/NavBot.cpp]: Dispensers do not

File: src/hacks/NavBot.cpp [hunk 15]
```diff
@@ -300,6 +824,95 @@ static bool stayNearPlayers(const bot_class_config &config, CachedEntity *&resul
...
+
+    static Timer wait_until_stay_near{};
+    if (current_task == task::engineer_task::staynear_engineer)
+    {
+        return true;
+    }
+    else if (wait_until_stay_near.test_and_set(1000))
+    {
+        // We're doing nothing? Do something!
+        return stayNearHelpers::stayNearPlayers(*config, last_target, last_area);
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/NavBot.cpp]: Oh. No i am not, thanks for the notice

File: src/hacks/NavBot.cpp [hunk 16]
```diff
@@ -643,6 +1268,18 @@ static slots getBestSlot(slots active_slot)
...
     }
+    case tf_engineer:
+    {
+        // Use wrench, because we are trying to build
+        if (current_task == task::engineer)
+        {
+            if (current_engineer_task == task::engineer_task::build_building)
+                return active_slot;
+            if (current_engineer_task == task::engineer_task::upgradeorrepair_building)
+                return melee;
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/NavBot.cpp]: We cannot repair/upgrade with a shotgun, and if we change to the shotgun while building we'll just get stuck indefinetly as the PDA counts as a different slot from elee

File: src/hooks/CreateMove.cpp (hunk 2 above)
Comment (BenCat07) [reply src/hooks/CreateMove.cpp]: Honestly, no idea

File: src/hooks/CreateMove.cpp (hunk 3 above)
Comment (BenCat07) [reply src/hooks/CreateMove.cpp]: I only set lerp on no-lerp activation now and undo on deactivation/unhook

File: src/hacks/Misc.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/Misc.cpp]: Fixed

File: src/hacks/MiscAimbot.cpp [hunk 17]
```diff
@@ -269,12 +273,242 @@ static void SapperAimbot()
...
+    {
+        // Upgrade lvel
+        int upgrade_level = CE_INT(ent, netvar.iUpgradeLevel);
+
+        // Don't upgrade mini sentries
+        if (CE_BYTE(ent, netvar.m_bMiniBuilding))
+            return false;
+
+        // Rvar to check
+        auto check_rvar = autoupgrade_sentry_level;
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/MiscAimbot.cpp]: Fixed

File: src/hacks/MiscAimbot.cpp [hunk 18]
```diff
@@ -269,12 +273,242 @@ static void SapperAimbot()
...
+
+    if (g_pLocalPlayer->clazz != tf_engineer)
+        return;
+
+    if (GetWeaponMode() != weaponmode::weapon_melee)
+        return;
+
+    bool first_run = false;
+    if (autorepair_priority)
+        first_run = true;
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/MiscAimbot.cpp]: Fixed by making code not-jank

File: src/hacks/MiscAimbot.cpp (hunk 11 above)
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/MiscAimbot.cpp]: Fixed by making code not-jank

File: src/hacks/MiscAimbot.cpp (hunk 10 above)
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/MiscAimbot.cpp]: Fixed by making code not-jank

File: src/hacks/MiscAimbot.cpp (hunk 9 above)
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/MiscAimbot.cpp]: Fixed

File: src/hacks/NavBot.cpp [hunk 19]
```diff
@@ -22,30 +25,82 @@ static settings::Int spy_ignore_time("navbot.spy-ignore-time", "5000");
...
-}
+Task current_task                   = task::none;
+engineer_task current_engineer_task = engineer_task::nothing;
+} // namespace task
+
 constexpr bot_class_config DIST_SPY{ 10.0f, 50.0f, 1000.0f };
 constexpr bot_class_config DIST_OTHER{ 100.0f, 200.0f, 300.0f };
 constexpr bot_class_config DIST_SNIPER{ 1000.0f, 1500.0f, 3000.0f };
+constexpr bot_class_config DIST_ENGINEER{ 600.0f, 1000.0f, 2500.0f };
+// Gunslinger Engineers really don't cate at all
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/NavBot.cpp]: Fixed

File: src/hacks/NavBot.cpp [hunk 20]
```diff
@@ -165,6 +368,315 @@ static bool navToSniperSpot()
...
+        // Nav to a Building spot
+        else if (navToBuildingSpot())
+        {
+            engineer_recheck.update();
+            return false;
+        }
+    }
+    // Metal
+    int metal = CE_INT(LOCAL_E, netvar.m_iAmmo + 12);
+    if ((dispenser_nav_timer.test_and_set(1000) && getDispenserHealthAndAmmo(metal)) || getHealthAndAmmo(metal))
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/NavBot.cpp]: Fixed

File: src/hacks/NavBot.cpp [hunk 21]
```diff
@@ -65,17 +120,41 @@ static void CreateMove()
...
+                    local_buildings.push_back(ent);
+            }
+            update_building_spots();
+        }
+    }
 
     if (get_health)
     {
-        if (getDispenserHealthAndAmmo())
+        int metal = 0;
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/NavBot.cpp]: Fixed

File: src/hacks/NavBot.cpp [hunk 22]
```diff
@@ -65,17 +120,41 @@ static void CreateMove()
...
+        if (engineer_mode)
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/NavBot.cpp]: Fixed

File: src/hacks/NavBot.cpp [hunk 23]
```diff
@@ -165,6 +368,315 @@ static bool navToSniperSpot()
...
+                        return true;
+                    }
+                    // We're too far away, go to building
+                    else if (navToBuilding())
+                        return true;
+            }
+            // If it's metal we're missing, get some metal
+            else if (metal < 100)
+            {
+                if ((dispenser_nav_timer.test_and_set(1000) && getDispenserHealthAndAmmo(metal)) || getHealthAndAmmo(metal))
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/NavBot.cpp]: Fixed

File: src/hacks/NavBot.cpp [hunk 24]
```diff
@@ -140,6 +221,128 @@ bool init(bool first_cm)
...
+        return area;
+    }
+    float second()
+    {
+        return min_distance;
+    }
+    std::vector<Vector *> third()
+    {
+        return enemy_list;
+    }
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/NavBot.cpp]: Honestly cannot remember why i did this

File: src/hacks/NavBot.cpp [hunk 25]
```diff
@@ -165,6 +368,315 @@ static bool navToSniperSpot()
...
+                current_task          = { task::engineer, priority };
+                current_engineer_task = task::engineer_task::goto_building;
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+static bool engineerLogic()
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/NavBot.cpp]: Yes

File: src/hacks/NavBot.cpp [hunk 26]
```diff
@@ -165,6 +368,315 @@ static bool navToSniperSpot()
...
+    if ((current_task != task::engineer || current_engineer_task == task::engineer_task::nothing || current_engineer_task == task::engineer_task::staynear_engineer) && current_task != task::health && current_task != task::ammo)
+    {
+        // Already have a building
+        if (local_buildings.size())
+        {
+            int metal = CE_INT(LOCAL_E, netvar.m_iAmmo + 12);
+            if (metal)
+                for (auto &building : local_buildings)
+                    // Hey hit the building thanks (gunslinger engineer shouldn't care)
+                    if (hacks::tf2::misc_aimbot::ShouldHitBuilding(building) && !HasWeapon(LOCAL_E, 142))
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/NavBot.cpp]: Done

File: src/hacks/NavBot.cpp [hunk 27]
```diff
@@ -165,6 +368,315 @@ static bool navToSniperSpot()
...
+                    {
+                        if (navToBuilding(building))
+                            return true;
+                    }
+
+            // Let's terrify some people (gunslinger engineer)
+            if (HasWeapon(LOCAL_E, 142))
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/NavBot.cpp]: done

File: src/hacks/NavBot.cpp [hunk 28]
```diff
@@ -674,7 +1311,36 @@ static void updateSlot()
...
+        auto it = std::find(local_buildings.begin(), local_buildings.end(), ent);
+        // If found, erase
+        if (it != local_buildings.end())
+            local_buildings.erase(it);
+    }
+};
+
+ObjectDestroyListener &listener()
+{
+    static ObjectDestroyListener object{};
```
[the remark this answers is not in the document]
Comment (BenCat07) [reply src/hacks/NavBot.cpp]: Code standard

File: src/MiscTemporary.cpp [hunk 29]
```diff
@@ -35,12 +41,33 @@ void color_callback(settings::VariableBase<int> &, int)
...
+static InitRoutine misc_init([]() {
     print_r.installChangeCallback(color_callback);
     print_g.installChangeCallback(color_callback);
     print_b.installChangeCallback(color_callback);
     no_zoom.installChangeCallback([](settings::VariableBase<bool> &, bool after) {
         static BytePatch patch(gSignatures.GetClientSignature, "81 EC ? ? ? ? A1 ? ? ? ? 8B 7D 08 8B 10 89 04 24 FF 92", 0x0, { 0x5B, 0x5E, 0x5F, 0x5D, 0xC3 });
         after ? patch.Patch() : patch.Shutdown();
     });
+    nolerp.installChangeCallback([](settings::VariableBase<bool> &, bool after) {
+        if (!after && backup_lerp)
```
Comment (TotallyNotElite) [inline src/MiscTemporary.cpp]: What if !after is true, but backup lerp is 0.0f? You should consider nesting the if statements instead.

File: src/MiscTemporary.cpp [hunk 30]
```diff
@@ -35,12 +41,33 @@ void color_callback(settings::VariableBase<int> &, int)
...
+            backup_lerp = cl_interp->GetFloat();
+            // We should adjust cl_interp to be as low as possible
+            if (cl_interp->GetFloat() > 0.152f)
+                cl_interp->SetValue(0.152f);
+        }
+    });
+    EC::Register(
+        EC::Shutdown,
+        []() {
+            cl_interp->SetValue(backup_lerp);
```
Comment (TotallyNotElite) [inline src/MiscTemporary.cpp]: no backup_lerp check

File: src/hacks/Spam.cpp (hunk 13 above)
Comment (TotallyNotElite) [inline src/hacks/Spam.cpp]: Header still not used anywhere.

File: src/helpers.cpp (hunk 1 above)
Comment (TotallyNotElite) [reply src/helpers.cpp]: We only have headers for a lot of things and yet don't crash. That's how c++ works. Have you verified if its really necessary to have these here?

Comment (BenCat07) [reply src/helpers.cpp]: Yes it is. The functions we call are either virtual or defined elsewhere. Else we crash. 100% of the time

File: src/MiscTemporary.cpp (hunk 29 above)
Comment (BenCat07) [reply src/MiscTemporary.cpp]: Fixed

File: src/MiscTemporary.cpp (hunk 30 above)
Comment (BenCat07) [reply src/MiscTemporary.cpp]: Fixed

File: src/hacks/Spam.cpp (hunk 13 above)
Comment (BenCat07) [reply src/hacks/Spam.cpp]: Fixed

**Comment by SylveonBottle** [comment]
Closes #683 and #684?

**Comment by TotallyNotElite** [comment]
Should probably. I'll let @BenCat07 do that tho in case there's something he still wants to do.

**Comment by EricSmith17** [comment]
fuuuuuuuuuuuuck youuuuuuuuuu

**Comment by TotallyNotElite** [comment]
No thanks, appreciate the effort tho.

**Pull request merged**